src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerExecutor.cs

```diff
@@ -803,15 +847,13 @@ private void ExecuteBlockActionsCore<TBlockStartAction, TBlockAction, TNodeActio
                     // Catch Exception from the start action.
                     ExecuteAndCatchIfThrows(
                         startAction.Analyzer,
-                        data =>
-                        {
-                            data.action(data.context);
-                            data.blockEndActions?.AddAll(data.scope.CodeBlockEndActions);
-                            data.syntaxNodeActions?.AddRange(data.scope.SyntaxNodeActions);
-                        },
-                        (action: codeBlockStartAction.Action, context: blockStartContext, scope: codeBlockScope, blockEndActions: codeBlockEndActions, syntaxNodeActions),
+                        static data => data.codeBlockStartAction.Action(data.blockStartContext),
+                        (codeBlockStartAction, blockStartContext),
                         new AnalysisContextInfo(Compilation, declaredSymbol, declaredNode),
                         cancellationToken);
+
+                    codeBlockEndActions!.AddAll(codeBlockScope.CodeBlockEndActions);
+                    syntaxNodeActions!.AddRange(codeBlockScope.SyntaxNodeActions);
```

Comment:
the ?.  was unnecessary.  we always have these values when going into these particular codepaths.

Reply:
note: the `!` will go away in the strong-typign PR as we will have a strong tie that if we get here, we must have made these temp collections to store the results in.